src/overlays/actors/ovl_Obj_Danpeilift/z_obj_danpeilift.c

```diff
@@ -28,35 +33,177 @@ const ActorInit Obj_Danpeilift_InitVars = {
-static InitChainEntry D_80C1B540[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80C1B540[];
-
-extern UNK_TYPE D_06000180;
-extern UNK_TYPE D_06000BA0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/func_80C1ADC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/ObjDanpeilift_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/ObjDanpeilift_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/func_80C1AF84.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/func_80C1AF94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/func_80C1B210.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/func_80C1B26C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/ObjDanpeilift_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Danpeilift/ObjDanpeilift_Draw.s")
+void ObjDanpeilift_UpdatePosition(ObjDanpeilift* this, s32 index) {
+    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->points[index]);
+}
+
+void ObjDanpeilift_Init(Actor* thisx, PlayState* play) {
+    Path* path;
+    ObjDanpeiliftActionFunc tempActionFunc;
+    ObjDanpeilift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.shape.rot.x = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &object_obj_danpeilift_Colheader_000BA0);
+    if (this->dyna.bgId == BG_ACTOR_MAX) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        this->speed = OBJDANPEILIFT_GET_SPEED(thisx);
+        if (this->speed < 0.0f) {
+            this->speed = -this->speed;
+        }
+        if (this->speed < 0.01f) {
+            this->actionFunc = ObjDanpeilift_DoNothing;
+        } else {
+            path = &play->setupPathList[OBJDANPEILIFT_GET_PATH(thisx)];
+            this->curPoint = OBJDANPEILIFT_GET_STARTING_POINT(thisx);
+            this->endPoint = path->count - 1;
+            this->direction = 1;
+            this->points = Lib_SegmentedToVirtual(path->points);
+            ObjDanpeilift_UpdatePosition(this, this->curPoint);
+            this->actionFunc = ObjDanpeilift_Move;
+        }
+    }
+}
+
+void ObjDanpeilift_Destroy(Actor* thisx, PlayState* play) {
+    ObjDanpeilift* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjDanpeilift_DoNothing(ObjDanpeilift* this, PlayState* play) {
+}
+
+void ObjDanpeilift_Move(ObjDanpeilift* this, PlayState* play) {
```

Comment:
Just want to note that a better match was recently found for ObjRaillift_Move that is getting changed in #944 (https://github.com/zeldaret/mm/blob/53b117874bb73e6a055fc075fadc9146e5c79126/src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c#L112). Could you update this to match that one?